### PR TITLE
Index write benchmark

### PR DESCRIPTION
**Pull request by bright-starry-sky**
This PR instead of #126.
Addressed panda-sheep's comment.

### CODE REVIEW

**Comment by bright-starry-sky** [comment]
Addressed comments; Resolved conflict.

File: src/storage/test/StorageIndexWriteBenchmark.cpp [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-#include "codec/RowWriterV2.h"
+#include "mock/MockCluster.h"
+#include "mock/MockData.h"
 #include <folly/Benchmark.h>
+#include "common/interface/gen-cpp2/storage_types.h"
+#include "common/interface/gen-cpp2/common_types.h"
+#include "utils/NebulaKeyUtils.h"
+#include "mock/AdHocIndexManager.h"
+#include "mock/AdHocSchemaManager.h"
+
```
Comment (panda-sheep) [inline src/storage/test/StorageIndexWriteBenchmark.cpp:20]: A small suggestion, put the header files of the same directory together

File: src/storage/test/StorageIndexWriteBenchmark.cpp [hunk 2]
```diff
@@ -24,85 +26,183 @@ DEFINE_string(root_data_path, "/tmp/IndexWritePref", "Engine data path");
...
+            break;
+        }
+        case IndexENV::INVALID_INDEX : {
+            indexMan->addTagIndex(spaceId, indexId, -1, mockTagIndexColumns());
+            break;
+        }
+        case IndexENV::MULITPLE_INDEX : {
+            indexMan->addTagIndex(spaceId, indexId, tagId, mockTagIndexColumns());
+            indexMan->addTagIndex(spaceId, indexId + 1, tagId, mockTagIndexColumns());
+            indexMan->addTagIndex(spaceId, indexId + 2, tagId, mockTagIndexColumns());
```
Comment (panda-sheep) [inline src/storage/test/StorageIndexWriteBenchmark.cpp:105]: I have a question:
three indexes are created here, but all three  indexes contain the same column,
In practice, such a case does not exist.  here only for test?

File: src/storage/test/StorageIndexWriteBenchmark.cpp [hunk 3]
```diff
@@ -119,9 +219,9 @@ void insertVertices(bool withoutIndex) {
             }
```
Comment (panda-sheep) [inline src/storage/test/StorageIndexWriteBenchmark.cpp:219]: line 103 ~ line 119 
This part of the code can be proposed separately
Avoid code duplication and Can be reused

File: src/storage/test/StorageIndexWriteBenchmark.cpp [hunk 4]
```diff
@@ -161,34 +261,37 @@ void insertVertices(bool withoutIndex) {
         }
```
Comment (panda-sheep) [inline src/storage/test/StorageIndexWriteBenchmark.cpp:261]: line 142 ~ line 161
This part of the code can be proposed separately
Avoid code duplication and Can be reused

File: src/storage/test/StorageIndexWriteBenchmark.cpp (hunk 2 above)
Comment (bright-starry-sky) [reply src/storage/test/StorageIndexWriteBenchmark.cpp:105]: > I have a question:
> three indexes are created here, but all three indexes contain the same column,
> In practice, such a case does not exist. Here only for test?

Good question, this test case used for write speed of the index.Although the structure of the index is the same, what is really written is three copies of data.

File: src/storage/test/StorageIndexWriteBenchmark.cpp [hunk 5]
```diff
@@ -414,60 +527,107 @@ unmatchIndex: Without match index, and through asyncAtomicOp.
...
 --total_vertices_size=2000000
 ============================================================================
 src/storage/test/StorageIndexWriteBenchmark.cpprelative  time/iter  iters/s
 ============================================================================
-withoutIndex                                                  3.00s  333.45m
-unmatchIndex                                                  4.72s  212.02m
-attachIndex                                                  40.56s   24.66m
-duplicateVerticesIndex                                       46.14s   21.67m
-multipleIndex                                               1.19min   14.00m
+withoutIndex                                                  7.06s  141.71m
```
Comment (darionyaphet) [inline src/storage/test/StorageIndexWriteBenchmark.cpp:535]: It looks like the performance has dropped by half ?

**Review by critical27 — APPROVED** [review]

**Pull request closed without merging**